### PR TITLE
Update golang to 1.23

### PR DESCRIPTION
**Pull request by tnozicka**
**Description of your changes:**
This PR updates golang to 1.23.

#### Requires
- [x] https://github.com/golang/go/issues/67089
- [x] https://github.com/scylladb/scylla-operator-release/pull/335

### CODE REVIEW

**Comment by tnozicka** [comment]
https://github.com/scylladb/scylla-operator-release/pull/335 landed
/retest

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/scylladb/scylla-operator
 
-go 1.22.3
+go 1.23
```
Comment (rzetelskik) [inline go.mod:3]: is removing the patch deliberate?

Comment (tnozicka) [reply go.mod:3]: yep, any 1.23 is good enough to build this at this point

**Pull request merged**